### PR TITLE
Add bargraph to Viz

### PR DESCRIPTION
**Pull request by audy**
- [x] plots a bargraph
- [ ] ~merge non top-n taxa into a single category (so that bar graphs add up to 1)~
- [x] consistency with the rest of the `viz` module
- [x] testing
- [x] documentation
- [ ] ~add support for `haxis` argument?~ moved to #215
- [ ] ~smart bar sorting: (if plotting at genus rank, sort bars by family first etc...)~
- [ ] ~xmax should not exceed 1.0 (or 100% of total reads)~

### CODE REVIEW

File: onecodex/analyses.py [hunk 1]
```diff
@@ -177,6 +177,9 @@ def to_df(self, rank='auto', top_n=None, threshold=None, remove_zeros=True, norm
         table_format : {'long', 'wide'}
             If wide, rows are classifications, cols are taxa, elements are counts. If long, rows are
             observations with three cols each: classification_id, tax_id, and count.
+        include_not_top_n : { True, False }
```
Comment (polyatail) [inline onecodex/analyses.py]: Did we end up not  using this kwarg?

File: onecodex/viz/_bargraph.py [hunk 2]
```diff
@@ -0,0 +1,124 @@
+from onecodex.exceptions import OneCodexException
+import altair as alt
```
Comment (polyatail) [inline onecodex/viz/_bargraph.py]: Let's try to keep the imports in two groups: xeno-modules at the top, then those from within `onecodex`, with an empty line between the two groups. And put them in alphabetical order. I might have messed this up elsewhere myself.

File: onecodex/viz/_bargraph.py [hunk 3]
```diff
@@ -0,0 +1,124 @@
...
+            Text label at the top of the plot.
+        xlabel : `string`, optional
+            Text label along the horizontal axis.
+        ylabel : `string`, optional
+            Text label along the vertical axis.
+        tooltip : `string` or `list`, optional
+            A string or list containing strings representing metadata fields. When a point in the
+            plot is hovered over, the value of the metadata associated with that sample will be
+            displayed in a modal.
+    """
```
Comment (polyatail) [inline onecodex/viz/_bargraph.py]: Can the docstring contents be inline with """ ?

File: onecodex/viz/_bargraph.py [hunk 4]
```diff
@@ -0,0 +1,124 @@
...
+
+        for f in tooltip:
+            df[magic_fields[f]] = magic_metadata[magic_fields[f]][
+                df["classification_id"]
+            ].tolist()
+
+        # add sample filenames
+        df["display_name"] = self.metadata["_display_name"][
+            df["classification_id"]
+        ].tolist()
```
Comment (polyatail) [inline onecodex/viz/_bargraph.py]: `_metadata_fetch` has a secret keyword where the string "Label" will be converted to the contents of `_display_name`. The benefit here is that the x axis title will then be "Label" if `xlabel=None`. Maybe add "Label" to `tooltip` and get rid of this?

File: onecodex/viz/_bargraph.py [hunk 5]
```diff
@@ -0,0 +1,124 @@
...
+
+        # add taxa names
+
```
Comment (polyatail) [inline onecodex/viz/_bargraph.py]: y u put newline here tho

File: onecodex/viz/_bargraph.py [hunk 6]
```diff
@@ -0,0 +1,124 @@
...
+        # about taxonomic ranks but as all taxonomic data will be coming from
+        # OCX this should be okay)
+        #
+
+        chart = (
+            alt.Chart(df)
+            .mark_bar()
+            .encode(
+                x=alt.X("display_name", axis=alt.Axis(title=xlabel)),
+                y=alt.Y("readcount_w_children", axis=alt.Axis(title=ylabel)),
```
Comment (polyatail) [inline onecodex/viz/_bargraph.py]: Let's check if `xlabel` or `ylabel` are `None` and provide a default behavior? I don't remember... if we set `title=None` does that mean there's no axis labels?

File: onecodex/viz/_heatmap.py [hunk 7]
```diff
@@ -41,6 +41,8 @@ def plot_heatmap(self, rank='auto', normalize='auto', top_n='auto', threshold='a
             A string or list containing strings representing metadata fields. When a point in the
             plot is hovered over, the value of the metadata associated with that sample will be
             displayed in a modal.
+        legend: `string`, optional
+            Variable to map the color scale to (default is 'abundance')
```
Comment (polyatail) [inline onecodex/viz/_heatmap.py]: 👍

File: onecodex/viz/_bargraph.py (hunk 6 above)
Comment (audy) [reply onecodex/viz/_bargraph.py]: Labels will be the same as if `axis=` wasn't specified. Altair is inconsistent with this. For example, setting `chart.properties(title=None)` will result in an error.

screen shot 2019-02-15 at 11 23 26 am

File: onecodex/analyses.py (hunk 1 above)
Comment (audy) [reply onecodex/analyses.py]: I was going to add this feature but decide that I should not include it in this PR. Thanks for catching.

File: onecodex/viz/_bargraph.py (hunk 4 above)
Comment (audy) [reply onecodex/viz/_bargraph.py]: set default `xaxis="Sample"`

**Pull request merged**